synapseclient/models/table.py

```diff
@@ -489,226 +909,2209 @@ def fill_from_dict(
         self.version_comment = synapse_table.get("versionComment", None)
         self.is_latest_version = synapse_table.get("isLatestVersion", None)
         self.is_search_enabled = synapse_table.get("isSearchEnabled", False)
-        self.columns = [
-            Column(id=columnId, name=None, column_type=None)
-            for columnId in synapse_table.get("columnIds", [])
-        ]
+
         if set_annotations:
             self.annotations = Annotations.from_dict(
                 synapse_table.get("annotations", {})
             )
         return self
 
-    @otel_trace_method(
-        method_to_trace_name=lambda _, **kwargs: f"Store_rows_by_csv: {kwargs.get('csv_path', None)}"
-    )
-    async def store_rows_from_csv_async(
-        self, csv_path: str, *, synapse_client: Optional[Synapse] = None
-    ) -> str:
-        """Takes in a path to a CSV and stores the rows to Synapse.
+    def to_synapse_request(self):
+        """Converts the request to a request expected of the Synapse REST API."""
+        entity = {
+            "name": self.name,
+            "description": self.description,
+            "id": self.id,
+            "etag": self.etag,
+            "parentId": self.parent_id,
+            "concreteType": concrete_types.TABLE_ENTITY,
+            "versionNumber": self.version_number,
+            "versionLabel": self.version_label,
+            "versionComment": self.version_comment,
+            "isSearchEnabled": self.is_search_enabled,
+            # When saving other (non-column) fields to Synapse we still need to pass
+            # in the list of columns, otherwise Synapse will wipe out the columns. We
+            # are using the last known columns to ensure that we are not losing any
+            "columnIds": [
+                column.id for column in self._last_persistent_instance.columns.values()
+            ]
+            if self._last_persistent_instance and self._last_persistent_instance.columns
+            else [],
+        }
+        delete_none_keys(entity)
+        result = {
+            "entity": entity,
+        }
+        delete_none_keys(result)
+        return result
+
+    async def upsert_rows_async(
+        self,
+        values: pd.DataFrame,
+        upsert_columns: List[str],
```

Comment:
discussion: I wonder if renaming this to "primary_keys" would be more synonymous with database language instead of introducing our own language.

Reply:
I agree with this renaming